ci/tests/general/general_alba.py

```diff
@@ -350,12 +352,27 @@ def filter_disks(disk_names, amount, disk_type):
             list_to_check = hdds.values()
 
         for disk_name in disk_names:
-            for disk in list_to_check:
-                if disk_name == disk['name']:
-                    filtered_disks.append(disk['name'])
-                    count += 1
-            if count == amount:
-                break
+            for node in list_to_check:
+                for disk in node.values():
+                    if disk_name == disk['name']:
+                        if disk['ip'] in sorted_disks:
```

Comment:
if disk['ip'] not in sorted_disks:
    sorted_disks[disk['ip']] = []
sorted_disks[disk['ip']].append...